src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8847,49 +8885,189 @@ void InitSurfacePropertyViewFactors(EnergyPlusData &state)
+                Surface.ViewFactorGroundIR = GroundSurfsViewFactor;
+                Surface.ViewFactorSkyIR = 1 - SrdSurfsViewFactor;
+                if (SrdSurfsNum > 0) {
+                    state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).IsSkyViewFactorSet = true;
+                    state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor = Surface.ViewFactorSkyIR;
+                }
+            } else {
+                // If neither ground nor sky view factor specified, continue to use the original proportion.
+                Surface.ViewFactorSkyIR *= 1 - SrdSurfsViewFactor;
+                Surface.ViewFactorGroundIR *= 1 - SrdSurfsViewFactor;
+                if (SrdSurfsNum > 0) {
+                    state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).IsSkyViewFactorSet = true;
+                    state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor = Surface.ViewFactorSkyIR;
+                    if (GndSurfsNum == 0) {
+                        state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor = Surface.ViewFactorGroundIR;
+                        state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).IsGroundViewFactorSet = true;
+                    }
+                }
+                if (GndSurfsNum > 0) {
+                    SetGroundViewFactorObject = true;
+                    state.dataSurface->GroundSurfsProperty(GndSurfsNum).IsGroundViewFactorSet = true;
+                }
+            }
+            if (SetGroundViewFactorObject) {
+                ReSetGroundSurfacesViewFactor(state, SurfNum);
+            }
         }
     }
 }
 
+void GetGroundSurfacesTemperatureAverage(EnergyPlusData &state)
+{
+    //  returns ground surfaces average temperature (deg C)
+    //  ground surfaces viewed by a building exterior surface
+    //  ground surfaces temperature weighed using view factors
+
+    // local vars
+    Real64 GndSurfaceTemp;
+    Real64 GndSurfViewFactor;
+    Real64 GndSurfaceTempSum;
+
+    if (!state.dataGlobal->AnyLocalEnvironmentsInModel) {
+        return;
+    }
+
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (!state.dataSurface->IsSurfPropertyGndSurfacesDefined(SurfNum)) continue;
+        auto &GndSurfsProperty = state.dataSurface->GroundSurfsProperty(state.dataSurface->GroundSurfsPropertyNum(SurfNum));
+        if (GndSurfsProperty.SurfsViewFactorSum == 0.0) {
+            state.dataSurface->UseSurfPropertyGndSurfTemp(SurfNum) = false;
+            continue;
+        }
+        GndSurfaceTemp = 0.0;
+        GndSurfViewFactor = 0.0;
+        GndSurfaceTempSum = 0.0;
+        for (int gSurfNum = 1; gSurfNum <= GndSurfsProperty.NumGndSurfs; gSurfNum++) {
+            GndSurfViewFactor = GndSurfsProperty.GndSurfs(gSurfNum).ViewFactor;
+            if (GndSurfViewFactor == 0.0) continue;
```

Comment:
I was going to suggest that this code is better if you just let the 0.0 pass through and the calculation work itself out, but if it is 0.0, it does avoid the Schedule lookup, so it's worth it to jump here.